presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/optimization/JdbcJoinPushdown.java

```diff
@@ -0,0 +1,132 @@
+import com.facebook.presto.spi.JoinTableInfo;
+import com.facebook.presto.spi.JoinTableSet;
+import com.facebook.presto.spi.TableHandle;
+import com.facebook.presto.spi.VariableAllocator;
+import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.plan.PlanNodeIdAllocator;
+import com.facebook.presto.spi.plan.TableScanNode;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.spi.ConnectorPlanRewriter.rewriteWith;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * JdbcJoinPushdown Optimizer:
+ * This Optimizer converts ConnectorHandleSet with various TableHandles to JdbcTableHandle which has list of all the JdbcTableHandles that can be pushed down.
+ * This results in Jdbc Join Pushdown.
+ *
+ * Example:
+ * Before Transformation:
+ *   --OutputNode
+ *   `-- TableScanNode
+ *       `-- Set<ConnectorTableHandle> (ConnectorHandleSet)
+ *          `-- TableHandle1
+ *          `-- TableHandle2
+ *          `-- TableHandle3
+ *
+ * After Transformation:
+ *   --OutputNode
+ *   `-- TableScanNode
+ *       `-- JdbcTableHandle
+ *           `-- List<ConnectorTableHandle>
+ *               `-- JdbcTableHandle1
+ *               `-- JdbcTableHandle2
+ *               `-- JdbcTableHandle3
+ */
+public class JdbcJoinPushdown
+        implements ConnectorPlanOptimizer
+{
+    @Override
+    public PlanNode optimize(PlanNode maxSubplan, ConnectorSession session, VariableAllocator variableAllocator, PlanNodeIdAllocator idAllocator)
+    {
+        return rewriteWith(new Rewriter(session, idAllocator), maxSubplan);
+    }
+
+    private class Rewriter
+            extends ConnectorPlanRewriter<Void>
+    {
+        private final ConnectorSession session;
+        private final PlanNodeIdAllocator idAllocator;
+
+        public Rewriter(ConnectorSession session, PlanNodeIdAllocator idAllocator)
+        {
+            this.session = requireNonNull(session, "session is null");
+            this.idAllocator = requireNonNull(idAllocator, "idAllocator is null");
+        }
+
+        @Override
+        public PlanNode visitTableScan(TableScanNode node, RewriteContext<Void> context)
+        {
+            TableHandle tableHandle = node.getTable();
+            if (tableHandle.getConnectorHandle() instanceof JoinTableSet) {
+                JoinTableSet tableHandles = (JoinTableSet) tableHandle.getConnectorHandle();
+                ImmutableList<JoinTableInfo> newJoinTableInfos = ImmutableList.copyOf(tableHandles.getInnerJoinTableInfos());
+
+                List<ConnectorTableHandle> newConnectorTableHandles = new ArrayList<>();
+                Map<VariableReferenceExpression, ColumnHandle> groupAssignments = new HashMap<>();
+                final String aliasPrefix = "T";
```

Comment:
Can you add a comment here -
`Generate aliases for each table being joined to avoid any ambuigious column name references. These aliases are used in QueryBuilder#buildSql`